store/snapshots/types/options.go

```diff
@@ -10,7 +12,12 @@ type SnapshotOptions struct {
 	KeepRecent uint32
 }
 
+// NewSnapshotOptions creates and returns a new SnapshotOptions instance.
+// It panics if the interval exceeds the maximum value for int64.
 func NewSnapshotOptions(interval uint64, keepRecent uint32) SnapshotOptions {
```

Comment:
is this tested?

Reply:
I did not add the test in the first place as this is legacy code and the constructor can easily be bypassed by the public attribute. Added now for regression 👍